server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -746,16 +774,29 @@ Batch nextBatch(SortedKeyValueIterator<Key,Value> iter, Range range, int num, Se
       iter.next();
     }
 
-    if (iter.hasTop() == false) {
-      endOfTabletReached = true;
-    }
+    if (yield.hasYielded()) {
+      continueKey = new Key(yield.getPositionAndReset());
+      skipContinueKey = true;
+      if (!range.contains(continueKey)) {
+        throw new IOException("Underlying iterator yielded to a position outside of its range: " + continueKey + " not in " + range);
+      }
 
-    if (endOfTabletReached) {
-      continueKey = null;
-    }
+      log.debug("Scan yield detected at position " + continueKey);
```

Comment:
Same here about trace instead of debug. This wouldn't help an operator figure out what's going on.